aerospike-core/src/batch/batch_executor.rs

```diff
@@ -0,0 +1,119 @@
+// Copyright 2015-2018 Aerospike, Inc.
+//
+// Portions may be licensed to Aerospike, Inc. under one or more contributor
+// license agreements.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"); you may not
+// use this file except in compliance with the License. You may obtain a copy of
+// the License at http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+// WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+// License for the specific language governing permissions and limitations under
+// the License.
+
+use std::cmp;
+use std::collections::HashMap;
+use std::sync::Arc;
+
+use crate::batch::BatchRead;
+use crate::cluster::partition::Partition;
+use crate::cluster::{Cluster, Node};
+use crate::commands::BatchReadCommand;
+use crate::errors::{Error, Result};
+use crate::policy::{BatchPolicy, Concurrency};
+use crate::Key;
+use futures::lock::Mutex;
```

Comment:
`tokio::sync::Mutex` is `fair`, it gives out locks in the order they were requested.
I don't know if this is significant, in this case.

Reply:
This actually seems to be a positive in the case of a data access library.